### DISCUSSION TITLE
gatsby-plugin-image missing fadeIn option

### DISCUSSION THREAD

**Discussion by lnikell** [gatsbyjs/gatsby#30699] [Ideas / Feature Requests]
## Description

We migrate from Gatsby v2 to v3, unfortunately, there are some issues that are coming along the way. Lighthouse will penalize your website if it has animation on images in the hero section especially, so in v2 we used fadeIn={false} to have better scores. However in v3 with `gatsby-plugin-image` it is not possible and the only way of overriding it is ugly CSS !important styles.

It would be nice to have that option in v3 to so it makes the migration process smoother and will result in better performance scores.

**Comment by kelvindecosta** [comment]
Hello!

The [`placeholder`](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image#placeholder) option when set to `"none"` fades the images in once they are loaded, without any layout shift.

I think it might help with the migration and performance scores.

**Comment by lnikell** [comment]
Hi @kelvindecosta, placeholder NONE does not remove fade in animations. There is no issues with layout shifting, animation affects on LCP.

**Reply by ascorbic** [reply]
Have you detected any actual changes to LCP with this? The fade-in is very short, and we've not detected anything significant in our own tests. Setting loading=eager will have more effect. If you can provide a repro that shows different scores then please share and I'll look into adding this as an option.

**Reply by lnikell** [reply]
Yes, @ascorbic. LCP metric counts everything that is happening in the viewport when the page is loading, animation adds extra 200ms to this. It is not huge but if we compare two versions with hero section image with 0ms vs 200ms animation, so the LCP will be better when it's 0ms animation.  

Also from a design perspective, it does not make any sense to have fadeIn effect especially on the hero section with big images, since it only visually slows downs the loading process.

I will try to prepare more data for you, thank you!

**Reply by AnatoliyKowalev** [reply]
@ascorbic Hello, we faced this problem too. 

We migrated to gatsby-plugin-image from gatsby-image and found out that if we set placeholder to NONE there is some extra fadein animation which makes smooth appearance. Previously we could disable any images appearance in "gatsby-image" by fadein={false}, but in "gatsby-plugin-image" it's unreal, there is no prop option for that. 

It's a problem because it loads two same images one of them used for animation but it double work for the network. That first image used for fadein has same size as original, it's not base64 format. 

Is there any chance to add fadeIn prop to GatsbyImage component like in "gatsby-image" plugin?

**Reply by ascorbic** [reply]
Thanks. We're looking into this.

**Reply by robmarshall** [reply]
Did this progress anywhere? I know that @ascorbic left Gatsby and he was originally driving this.

**Reply by ocano-diego** [reply]
Any updates around here? :)

**Reply by MANTENN** [reply]
Fortunately, the animation is done with CSS--with the help of JS. 

Set the opacity to 1 and set no transition on elements within the container class that have the eager loading attribute. 

This is the selector
```css
.gatsby-image-wrapper [data-main-image] {
```

**Reply by sn3h** [reply]
this is not a solution, just temporary hack.. the css selector may change and u'll never know

 :( fadeIn needs to be boolean and fadeInSpeed reintroduced, or turned off completely, at least for placeholder : none

**Reply by allankiezel** [reply]
Bumping this. The fade-in is pretty brutal on above the fold images and does have an impact on google pagespeed.

**Reply by tlehren** [reply]
Bumping as well, is there any solution?

**Reply by robmarshall** [reply]
Not too fussed about stopping the animation CSS. I would like to remove the base64 weight from the page completely - never have it added.

**Comment by DennisKraaijeveld** [comment]
Just wanted to let you guys know that I also have been struggling with LCP and I thought (and hoped) that Gatsby Image was the cause of this. I think it is really the easy was to just think it's the image and I also thought this was my main problem. So I did tried all the tips I did read about here and on other places, like using the `.gatsby-image-wrapper [data-main-image]  {` selector. But things got worse and worse.

My LCP was round 4-5s. I did some code splitting, removed the css selector and just let Gatsby Image do her work and kaboom, 0,8s LCP. Still need some other improvements btw..

![image](https://user-images.githubusercontent.com/50783193/148564204-06f5fc26-4823-475e-a8f9-1f4f1f4f0a20.png)

**Reply by DuneArchitect** [reply]
> Just wanted to let you guys know that I also have been struggling with LCP and I thought (and hoped) that Gatsby Image was the cause of this. I think it is really the easy was to just think it's the image and I also thought this was my main problem. So I did tried all the tips I did read about here and on other places, like using the `.gatsby-image-wrapper [data-main-image] {` selector. But things got worse and worse.
> 
> My LCP was round 4-5s. I did some code splitting, removed the css selector and just let Gatsby Image do her work and kaboom, 0,8s LCP. Still need some other improvements btw..

How did you do your code splitting?  Could you link to any resources on this

**Reply by dev-searchengineunity** [reply]
I would also like to know as well. Thanks in advance!

**Comment by MikeBroders** [comment]
hey, for me it's less the effect to core vitals... it's more the visual effect when you need to use rehydration... the page visibility loads twice. without the fade in effect of GatsbyImage at least you wouldn't recognise it. so a fade option would be really helpful.

**Comment by viidn** [comment]
Any update around here?
